spec/allocator_spec.rb

```diff
@@ -53,27 +53,173 @@
       expect { new_allocator(1).allocate([rate, 1 - rate]) }.not_to raise_error
     end
 
+    specify "#allocate raise ArgumentError when invalid strategy is provided" do
+      expect { new_allocator(0.03).allocate([0.5, 0.5], :bad_strategy_name) }.to raise_error(ArgumentError, "Invalid strategy. Valid options: :roundrobin, :roundrobin_reverse, :nearest")
+    end
+
+    specify "#allocate raises an error when splits exceed 100%" do
+      expect { new_allocator(0.03).allocate([0.5, 0.6]) }.to raise_error(ArgumentError, "splits add to more than 100%")
```

Comment:
Money has another method called `split`, I think we should avoid that terminology here
```suggestion
      expect { new_allocator(0.03).allocate([0.5, 0.6]) }.to raise_error(ArgumentError, "allocations add to more than 100%")
```